samples/matter/common/src/bridge/bridged_device_data_provider.h

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#pragma once
+
+#include <app/util/attribute-storage.h>
+
+class BridgedDeviceDataProvider {
+public:
+	using UpdateAttributeCallback = void (*)(BridgedDeviceDataProvider &dataProvider, chip::ClusterId clusterId,
+						 chip::AttributeId attributeId, void *data, size_t dataSize);
+
+	explicit BridgedDeviceDataProvider(UpdateAttributeCallback callback) { mUpdateAttributeCallback = callback; }
```

Comment:
I realized we missed virtual destructor here resulting in memory leaks when destroying child objects. I can fix it the follow up PR however.

Reply:
Please fix it in your PR, since we should finish this PR to unblock you.